### PR TITLE
solution

### PR DESCRIPTION
**Pull request by RMasnyuk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (gestap) [inline .gitignore]: the document must end with one empty line

File: Numberdrome.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+class Numberdrome {
+    constructor() {
```
Comment (gestap) [inline Numberdrome.js]: use 2 whitespaces instead of 4

Comment (gestap) [reply Numberdrome.js]: https://github.com/mate-academy/style-guides/blob/master/javascript.md#whitespace--spaces

File: Numberdrome.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    addNumber(n) {
+        this.numbers.push(n);
+    }
+
+    removeNumber(n) {
+        this.numbers.splice(this.numbers.indexOf(n), 1);
+    }
+
+    sum() {
+        return this.numbers.length === 0 ? 0 : this.numbers.reduce((sum, current) => {
```
Comment (gestap) [inline Numberdrome.js]: no need for a ternary operator, you have initialValue in `reduce`

File: Numberdrome.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return sum + current;
+        }, 0);
+    }
+
+    product() {
+        return this.numbers.length === 0 ? 1 : this.numbers.reduce((prod, current) => {
```
Comment (gestap) [inline Numberdrome.js]: no need for a ternary operator, you have initialValue in `reduce`

File: Numberdrome.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }, 1);
+    }
+    min() {
+        try {
+            if (!this.numbers.length) {
+                throw 'list is empty!'
+            }
+            return Math.min(...this.numbers);
+        } catch (e) {
+            console.log(e);
```
Comment (gestap) [inline Numberdrome.js]: you need to `throw an error` instead return undefined

File: Numberdrome.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    max() {
+        try {
+            if (!this.numbers.length) {
+                throw 'list is empty!'
+            }
+            return Math.max(...this.numbers);
+        } catch (e) {
+            console.log(e);
```
Comment (gestap) [inline Numberdrome.js]: you need to `throw an error` instead return undefined

**Review by ost-ptk — CHANGES_REQUESTED** [review]

**Review by NclsVI — APPROVED** [review]

**Pull request closed without merging**